saleor/checkout/core.py

```diff
@@ -28,6 +29,8 @@ def __init__(self, cart, user, tracking_code):
         self.tracking_code = tracking_code
         self.user = user
         self.discounts = cart.discounts
+        self.cached_shipping_method = None
```

Comment:
I'm not sure if `cached_` prefix object fits here best. This cache is an implementation detail and it won't be never exposed as an public part of Checkout object API, so I recommend to rename these parameters to private fields like `self._shipping_method_cache` and `self._shipping_address_cache`